sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/PersonalizerAsyncClient.java

```diff
@@ -0,0 +1,310 @@
+// Licensed under the MIT License.
+
+package com.azure.ai.personalizer;
+
+import com.azure.ai.personalizer.implementation.PersonalizerClientV1Preview3Impl;
+import com.azure.ai.personalizer.implementation.util.Transforms;
+import com.azure.ai.personalizer.models.PersonalizerRankMultiSlotOptions;
+import com.azure.ai.personalizer.models.PersonalizerRankMultiSlotResult;
+import com.azure.ai.personalizer.models.PersonalizerRankOptions;
+import com.azure.ai.personalizer.models.PersonalizerRankResult;
+import com.azure.ai.personalizer.models.PersonalizerRewardMultiSlotOptions;
+import com.azure.ai.personalizer.models.PersonalizerRewardOptions;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * This class provides an asynchronous client that contains the operations that apply to Azure Personalizer.
+ * Operations allowed by the client are ranking a set of actions, activation and sending rewards for a single
+ * slot as well as multi-slot scenarios.
+ *
+ * <p><strong>Instantiating an asynchronous Personalizer Client</strong></p>
+ * <p>
+ * <!-- src_embed com.azure.ai.personalizer.PersonalizerAsyncClient.instantiation -->
+ * <pre>
+ * PersonalizerAsyncClient personalizerClient = new PersonalizerClientBuilder&#40;&#41;
+ *     .credential&#40;new AzureKeyCredential&#40;&quot;&#123;key&#125;&quot;&#41;&#41;
+ *     .endpoint&#40;&quot;&#123;endpoint&#125;&quot;&#41;
+ *     .buildAsyncClient&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.ai.personalizer.PersonalizerAsyncClient.instantiation -->
+ *
+ * @see PersonalizerClientBuilder
+ */
+@ServiceClient(builder = PersonalizerClientBuilder.class, isAsync = true)
+public final class PersonalizerAsyncClient {
+
+    private final ClientLogger logger = new ClientLogger(PersonalizerAsyncClient.class);
+    private final PersonalizerClientV1Preview3Impl service;
+
+    PersonalizerAsyncClient(PersonalizerClientV1Preview3Impl service) {
+        this.service = service;
+    }
+
+    /**
+     * Request a list of actions to be ranked.
+     *
+     * <p>Submit a Personalizer rank request. Receives a context and a list of actions. Returns which of the provided
+     * actions should be used by your application, in rewardActionId.
+     *
+     * @param rankRequest A Personalizer Rank request.
+     * @return returns which action to use as rewardActionId, and additional information about each action as a result
+     * of a Rank request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
```

Comment:
Need to add throws IllegalArgumentException

Reply:
Not sure, if this is resolved. This method javadoc doesn't include throws?

Reply:
Fixed this time.